Create defchem.ttl

maschu09 opened the pull request:
- uploaded version of defchem.ttl at 35903587d83239e35a42a805e0b291098bde76d1
  - https://test.wmocodes.info/def/chemistry
  - [ ] confirmation of /def/chemistry as the agreed location
  - [ ] file has been changed since this, please reload
- uploaded version of medium.ttl at 991e82a4a5db8141e009312c2b6631c7cc4f6ae3
  - https://test.wmocodes.info/common/medium
  - [ ] confirmation of /common/medium as the agreed location
  - [ ] file has been changed since this, please reload

maschu09 committed:
Create defchem.ttl

marqh commented on ttl/defchem.ttl:
```diff
@@ -0,0 +1,17 @@
+@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix skos:  <http://www.w3.org/2004/02/skos/core#> .
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+@prefix defchem: <https://ontology.geodab.eu/def/chemistry/>
```
we need to use a wmo definition, i suggest

```
<https://test.wmocodes.info/def/chemistry/>
```

maschu09 committed:
Update defchem.ttl

changed base url to wmo.codes.int

marqh committed:
Update defchem.ttl

marqh committed:
Update defchem.ttl

maschu09 committed:
Merge pull request #1 from marqh/patch-2

Update defchem.ttl

marqh commented on ttl/defchem.ttl:
```diff
@@ -0,0 +1,27 @@
+@prefix rdfs:  <http://www.w3.org/2000/01/rdf-schema#> .
+@prefix skos:  <http://www.w3.org/2004/02/skos/core#> .
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+
```
need to add

```
@prefix ldp:   <http://www.w3.org/ns/ldp#> .
```

maschu09 committed:
Update defchem.ttl

added prefix ldp

marqh commented on ttl/defchem.ttl:
```diff
@@ -0,0 +1,28 @@
+@prefix skos:  <http://www.w3.org/2004/02/skos/core#> .
+@prefix owl:   <http://www.w3.org/2002/07/owl#> .
+@prefix ldp:   <http://www.w3.org/ns/ldp#> .
+
```
```
@prefix reg:   <http://purl.org/linked-data/registry#> .
```

maschu09 committed:
Update defchem.ttl

added prefix reg

maschu09 committed:
Update defchem.ttl

corrected a typo

maschu09 committed:
Create medium.ttl

marqh commented:
uploaded version of defchem.ttl at
https://test.wmocodes.info/def/chemistry
- [ ] file has been changed since this, please reload

maschu09 committed:
Create chemical_species.ttl

maschu09 committed:
Create wmdsObservedVariable.ttl

maschu09 committed:
Update medium.ttl

removed prefix and empty "broader" relations

maschu09 commented:
@marqh Please enjoy your weekend and look at new files early next week ;-)

maschu09 committed:
Create atmospheric_chemical_species.ttl

maschu09 commented:
@marqh I uploaded atmospheric_chemical_species.ttl with all the corrections we discussed yesterday. Note that I added "atmospheric" to the file name, because I realized that scientists from other disciplines (e.g. minearology) might need a very different set of species, and I don't believe it will be possible to create and maintain one list for everyone. It may therefore be better to establish separate lists for different disciplines and link the contents via exactMatch or similarMatch relations. Please try to upload this to common/atmospheric-chemical-species.

maschu09 commented:
@marqh A general question: while preparing the Excel "master" spreadsheet I thought that this information would be much better maintained in a database. But in my understanding the registry is already a database, is it not? So, woul dit actually make sense to set up our own "master" database here, or wouldn't it be a better workflow if - once we have populated the registry - we would have tools to expand, edit, or deprecate the registry's content. I would defintively like to get away from spreadsheets, but the fewer systems are involved the less chances there are for inconcistencies.

maschu09 committed:
Create wmdObservedVariable.ttl

maschu09 commented:
@marqh OK - done it! wmdObservedVariable.ttl also ready for upload from my perspective. Should also go into common.

marqh commented:
Hello @maschu09 

line 391 in 
atmospheric_chemical_species.ttl
https://github.com/maschu09/code-manuals/blob/57fd51755fbcf991e8feb3a19c01aad943f89492/ttl/atmospheric_chemical_species.ttl#L391
(and many other lines)

contain an element of the form:

```
 skos:closeMatch gemet:concept?cp=623 ;
```

The `?`, `%`, `-` and `=` are special characters in TTL, and must be 'escaped', as per
https://www.w3.org/TR/turtle/#sec-grammar
(I think)

So, 

```
 skos:closeMatch gemet:concept\?cp\=623 ;
```

etc.

the nist URIs are failing to parse, I'm trying to unpick the special characters in these but I can't get to the bottom of the problem

Is it easy for you to reverse the implementation of the nist prefix and include all of these as full URIs enclosed in `<>` then I think this might be a simpler approach.  It seems regressive to me, but I think it may be pragmatic
i think the same is true of the gmet and ofmpub ones

although the prefixing is a useful encoding, it turns out in these cases it is causing more problems than it solves.
As it was my suggestion, I can only apologise for this

marqh commented:
I have done a local search and replace and uploaded content to:
https://test.wmocodes.info/common/chemical-species

please let me know when you have altered your creation process to handle these properly

marqh commented:
wmdsObservedVariable requires

```
@prefix defchem: <http://codes.wmo.int/def/defchem/> .
@prefix medium: <http://codes.wmo.int/common/medium/> .
@prefix chemical-species: <http://codes.wmo.int/common/chemical-species/> .
```

at the top

additionally, a number of the chemical species identifiers have commas in their name, which needs to be 'escaped' in ttl
e.g. line 2968

```
        defchem:chemical-species chemical-species:1\,2-dichlorotetrafluoroethane ;
```

maschu09 committed:
Delete wmdsObservedVariable.ttl

fixing syntactic errors (escape commata) and add prefixes for chemcial-species and medium

maschu09 committed:
Create wmdObservedVariable.ttl

maschu09 committed:
Update atmospheric_chemical_species.ttl

File produced again from excel spreadsheet; replaced prefixes for foreign thesauri with explicit web links.

maschu09 commented:
@marqh I have uploaded two new ttl files for atmospheric_chemical_species.ttl and wmdObservedVariables.ttl in which the issues you raised have been fixed.

marqh commented:
> @marqh I have uploaded two new ttl files for atmospheric_chemical_species.ttl and wmdObservedVariables.ttl in which the issues you raised have been fixed.

I can confirm that these files now parse and have been uploaded to 
http://test.wmocodes.info/common/chemical-species
and
http://test.wmocodes.info/common/wmdsObservedVariable
